dashboard-api/internal/platform/server/handler/accounts/create.go

```diff
@@ -0,0 +1,82 @@
+package accounts
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	studiesmanager "github.com/vlab-research/vlab/dashboard-api/internal"
+	"github.com/vlab-research/vlab/dashboard-api/internal/platform/server/middleware/auth"
+	"github.com/vlab-research/vlab/dashboard-api/internal/platform/storage"
+)
+
+type createResponse struct {
+	Data *studiesmanager.Account `json:"data"`
+}
+
+// CreateHandler is a gin handler that is used to create
+// a new account object in the database
+func CreateHandler(repositories storage.Repositories) gin.HandlerFunc {
+	return func(ctx *gin.Context) {
+		var a, err = parseRequest(ctx)
+		if err != nil {
+			ctx.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+			return
+		}
+
+		a.UserId = auth.GetUserIdFrom(ctx)
+		err = repositories.Account.Create(
+			ctx,
+			a,
+		)
+
+		if err != nil {
+			ctx.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+			return
+		}
+
+		ctx.JSON(http.StatusCreated, createResponse{Data: a})
+	}
+}
+
+// parseRequest is a simple parser for the accounts endpoint
+// that also handles payload validation
+func parseRequest(ctx *gin.Context) (*studiesmanager.Account, error) {
+	var a *studiesmanager.Account
+	if err := ctx.BindJSON(&a); err != nil {
+		fmt.Printf("%s", err)
+		return &studiesmanager.Account{}, err
+	}
+
+	if err := validateCredentials(a); err != nil {
+		return &studiesmanager.Account{}, err
+	}
+
+	return a, nil
+}
+
+// validateCredentials is a simple validator used to handle the unstructured
+// nature of account credentials
+func validateCredentials(a *studiesmanager.Account) error {
```

Comment:
If we are anyways having _some_ of the structure and validating, I guess I'm not convinced it isn't just as easy to have _all_ the structure and use Go Types, which is also easy to read and can be shared elsewhere, no? 

i.e. given we're all in one Go module, we have a struct called `FlyCredentials` already in the Fly connector. Why not just use that? We can just try to marshal the json into all the supported types and if it fails, throw an error.

Reply:
Yeah happy to do this! Was thinking it might be better, we could also potentially use generics for the creation then :thinking: